tests/kernel/mutex/mutex_null_ptr_deref/src/main.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022 SeediqQ from Issue 48056
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+
+#define TIMEOUT_MS 100
+#define STACKSZ	   8192
+
+static struct k_mutex mutex;
+static struct k_thread thread;
+
+K_THREAD_STACK_DEFINE(stack, STACKSZ);
+
+static void test_thread(void *p1, void *p2, void *p3)
+{
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	k_mutex_lock(&mutex, K_MSEC(TIMEOUT_MS));
+
+	k_mutex_unlock(&mutex);
+}
+
+ZTEST(mutex_timeout_null_ptr_deref, test_mutex_timeout_error)
+{
+	k_mutex_init(&mutex);
+
+	k_thread_create(&thread, stack, K_THREAD_STACK_SIZEOF(stack), test_thread, NULL, NULL, NULL,
+			8, 0, K_NO_WAIT);
```

Comment:
I recommend trying to align to a tick boundary prior to calling k_thread_create(). As both test_thread() and test_mutex_timeout_error() delay for TIMEOUT_MS, this should help ensure that they both expire on the same tick.

Reply:
The simplest way is sleep for a tick.

Reply:
Yeah, k_usleep(1) and k_sleep(K_TICKS(1)) are patterns that appear again and again in timer-related tests.

Reply:
Oh, but this is a backport, so the ship already sailed :)  No reason not to fix, but would have to be done in main.

Reply:
Or no, it's not, just can't read my labels today :)